sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -119,7 +119,7 @@ internal EventProcessorStatus Status
                 {
                     try
                     {
-                        if (!ProcessorRunningSync.Wait(100))
+                        if (!ProcessorRunningGuard.Wait(100))
```

Comment:
Curious - Why did we choose a different timeout here vs what we used in our derived class?

Reply:
Different purposes; this was a specific, if somewhat imperfect, attempt to resolve a race condition where we were potentially seeing an outdated status.   In general, if we don't acquire the semaphore here quickly, we can make reasonably infer the state.   Because this is a property that can potentially be read in a loop or other production scenario where it is assumed "a property is lightweight", we felt it was important to minimize the potential wait.

In the processor client, the intent was to guard against multiple calls that manipulate the state of the processor by adding/removing handlers.   Since that is an operation that can't be done when the processor is running, there isn't an expectation that "this is lightweight enough to call often" and usage scenarios are limited to only those where processor configuration is being adjusted (initialization, recovery, some kind of change to the host environment) and we felt that it was reasonable to allow a bit more time for things to resolve.

Reply:
Thanks for the explanation!